internal/interceptors/auth.go

```diff
@@ -68,9 +72,25 @@ func secureCompare(expected, actual string) bool {
 	return false
 }
 
+// extractProjectIDFromURI : This function extracts the projectID from the HTTP request URI.
+// Example1: extractProjectIDFromURI("/v1/projects/project1/topics/t123") = project1
+// Example2: extractProjectIDFromURI("/v1/projects/project7/subscriptions/s123") = project7
+// Example3: extractProjectIDFromURI("/v1/admin/topic/t987") = ""
```

Comment:
will this middleware be called for admin routes ? example 3 doesn't seem to be valid. this middleware should be only for client auth. admin auth can be a different middleware

Reply:
Middlewares are different. Both of them are using a common function to extract `getUserPasswordProjectID()` from the `authorization` header. This is needed for both the auth types.